tests/multi_gpu/core/test_fairscale_megatron.py

```diff
@@ -137,7 +133,69 @@ def test_forward_hooks_FairscaleLayers():
         "row_linear": (None, None, None),
     }
     for module_name, expected_shape in hook_functions.items():
-        model.register_hook_function(HookFunction(module_name, expected_shape))
+        model.register_forward_hook(HookFunction(module_name, expected_shape))
+
+    _, _ = model(inputs)
+
+    if dist.get_rank() == 0:
+        assert torch.allclose(
+            output_dict["vocab_parallel_embedding"],
+            output_dict["vocab_embedding"],
+            atol=1e-7,
+        )
+        assert torch.allclose(
+            output_dict["parallel_embedding"], output_dict["embedding"], atol=1e-7
+        )
+        assert torch.allclose(
+            output_dict["column_parallel_linear"], output_dict["col_linear"], atol=1e-7
+        )
+        assert torch.allclose(
+            output_dict["row_parallel_linear"], output_dict["row_linear"], atol=1e-7
+        )
+        logger.info("Tests successful.")
+
+    Utils.destroy_model_parallel()
+
+
+@register_fairscale_megatron_test
+def test_gpu_offload_FairscaleLayers():
+    Utils.initialize_model_parallel(2, 1, 2)
```

Comment:
Maybe create a utility similar to `pytest.mark.parametrize(...)` for usage with the `submitit` multi-gpu tests.